include/mun/runtime.h

```diff
@@ -0,0 +1,275 @@
+    constexpr MunGuid type_guid<double>() {
+        return MunGuid{
+            { 143, 142, 214, 135, 13, 131, 110, 33, 172, 233, 214, 19, 161, 21, 42, 130 }
+        };
+    }
+
+    template<>
+    constexpr const char* type_name<double>() {
+        return "@core::float";
+    }
+
+    template<>
+    constexpr MunGuid type_guid<int64_t>() {
+        return MunGuid{
+            { 143, 152, 209, 84, 128, 137, 239, 52, 175, 82, 21, 160, 15, 102, 47, 114 }
+        };
+    }
+
+    template<>
+    constexpr const char* type_name<int64_t>() {
+        return "@core::int";
+    }
+
+    template<>
+    constexpr MunGuid type_guid<void>() {
+        return MunGuid{
+            { 202, 82, 158, 158, 79, 247, 42, 14, 1, 174, 64, 173, 40, 117, 124, 68 }
+        };
+    }
+
+    template<>
+    constexpr const char* type_name<void>() {
+        return "@core::empty";
+    }
+
+    /** A wrapper around a `MunErrorHandle`.
+     *
+     * Frees the corresponding error object on destruction, if it exists.
+     */
+    class Error {
+    public:
+        /** Constructs an error from a `MunErrorHandle`.
+         *
+         * \param handle an error handle
+         */
+        Error(MunErrorHandle handle) noexcept
+            : m_handle(handle) {}
+
+        /** Move constructs an error.
+         *
+         * \param other an rvalue reference to an error
+         */
+        Error(Error&& other) noexcept
+            : m_handle(other.m_handle) {
+            other.m_handle._0 = 0;
+        }
+
+        /** Destructs the error. */
+        ~Error() noexcept {
+            mun_error_destroy(m_handle);
+        }
+
+        /** Retrieves the error message, if it exists, otherwise returns a nullptr.
+         *
+         * The message is UTF-8 encoded.
+         */
+        const char* message() noexcept {
+            return mun_error_message(m_handle);
+        }
+
+        /** Retrieves whether an error exists */
+        operator bool() const noexcept {
+            return m_handle._0 != 0;
+        }
+
+    private:
+        MunErrorHandle m_handle;
+    };
+
+    /** A wrapper around a `MunRuntimeHandle`.
+     *
+     * Frees the corresponding runtime object on destruction, if it exists.
+     */
+    class Runtime {
+        friend std::optional<Runtime> make_runtime(std::string_view library_path) noexcept;
+        
+        /** Constructs a runtime from an instantiated `MunRuntimeHandle`.
+         *
+         * \param handle a runtime handle
+         */
+        Runtime(MunRuntimeHandle handle) noexcept
+            : m_handle(handle)
+        {}
+
+    public:
+        /** Move constructs a runtime
+         *
+         * \param other an rvalue reference to a runtime
+         */
+        Runtime(Runtime&& other) noexcept
+            : m_handle(other.m_handle) {
+            other.m_handle._0 = nullptr;
+        }
+
+        /** Destructs a runtime */
+        ~Runtime() noexcept {
+            mun_runtime_destroy(m_handle);
```

Comment:
Wont this fail if the handle was moved?

Reply:
`mun_runtime_destroy` nicely handles `nullptr`